src/gui/sdlmain.cpp

```diff
@@ -2550,12 +2559,113 @@ static void update_frame_texture([[maybe_unused]] const uint16_t *changedLines)
 	}
 }
 
+static std::optional<RenderedImage> get_rendered_output_from_backbuffer()
+{
+	RenderedImage image = {};
+
+	auto allocate_image = [&]() {
+		image.width              = sdl.clip.w;
+		image.height             = sdl.clip.h;
+		image.double_width       = false;
+		image.double_height      = false;
+		image.flip_vertical      = false;
+		image.pixel_aspect_ratio = {1};
+		image.bits_per_pixel     = 24;
+		image.pitch              = image.width * (image.bits_per_pixel / 8);
+		image.palette_data       = nullptr;
+
+		uint8_t* image_data = static_cast<uint8_t*>(
+				std::malloc(image.height * image.pitch));
+		image.image_data = image_data;
```

Comment:
Suggest directly assigning and `new[]`ing:

``` c++
image.image_data = new uint8_t[image.height * image.pitch];
```

This will match up with the `new`s used in the `RenderedImage` struct, where this one and the `RenderedImage` members are `delete`d.

Reply:
Sure. From memory, had to do it this way because I tried to make `image_data` in the struct const. But I gave up after a while, it just generated too many warnings and it necessitated too much const juggling for not much benefit.

If you use pointers, just be careful, that's it 🤷🏻